server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -0,0 +1,502 @@
+        indexResult.setTranslogLocation(location);
+        indexResult.setTook(System.nanoTime() - index.startTime());
+        indexResult.freeze();
+        localCheckpointTracker.advanceMaxSeqNo(index.seqNo());
+        return indexResult;
+    }
+
+    @Override
+    public DeleteResult delete(Delete delete) throws IOException {
+        ensureOpen();
+        DeleteResult deleteResult = new DeleteResult(delete.version(), delete.primaryTerm(), delete.seqNo(), true);
+        final Translog.Location location = translog.add(new Translog.Delete(delete, deleteResult));
+        deleteResult.setTranslogLocation(location);
+        deleteResult.setTook(System.nanoTime() - delete.startTime());
+        deleteResult.freeze();
+        localCheckpointTracker.advanceMaxSeqNo(delete.seqNo());
+        return deleteResult;
+    }
+
+    @Override
+    public NoOpResult noOp(NoOp noOp) throws IOException {
+        ensureOpen();
+        NoOpResult noOpResult = new NoOpResult(noOp.primaryTerm(), noOp.seqNo());
+        final Translog.Location location = translog.add(new Translog.NoOp(noOp.seqNo(), noOp.primaryTerm(), noOp.reason()));
+        noOpResult.setTranslogLocation(location);
+        noOpResult.setTook(System.nanoTime() - noOp.startTime());
+        noOpResult.freeze();
+        localCheckpointTracker.advanceMaxSeqNo(noOp.seqNo());
+        return noOpResult;
+    }
+
+    @Override
+    public GetResult get(Get get, BiFunction<String, SearcherScope, Searcher> searcherFactory) throws EngineException {
+        return getFromSearcher(get, searcherFactory, SearcherScope.EXTERNAL);
+    }
+
+    @Override
+    protected ReferenceManager<OpenSearchDirectoryReader> getReferenceManager(SearcherScope scope) {
+        return readerManager;
+    }
+
+    @Override
+    public boolean isTranslogSyncNeeded() {
+        return translog.syncNeeded();
+    }
+
+    @Override
+    public boolean ensureTranslogSynced(Stream<Translog.Location> locations) throws IOException {
+        boolean synced = translog.ensureSynced(locations);
+        if (synced) {
+            translog.trimUnreferencedReaders();
+        }
+        return synced;
+    }
+
+    @Override
+    public void syncTranslog() throws IOException {
+        translog.sync();
+        translog.trimUnreferencedReaders();
+    }
+
+    @Override
+    public Closeable acquireHistoryRetentionLock() {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public Translog.Snapshot newChangesSnapshot(
+        String source,
+        long fromSeqNo,
+        long toSeqNo,
+        boolean requiredFullRange,
+        boolean accurateCount
+    ) throws IOException {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Deprecated
+    @Override
+    public Translog.Snapshot newChangesSnapshotFromTranslogFile(String source, long fromSeqNo, long toSeqNo, boolean requiredFullRange)
```

Comment:
@mch2 I added this implementation temporarily, let me know if this is good or you just want to return null or entirely remove this method from Engine as it will be deprecated

Reply:
Removing the method is a breaking change. We can't introduce that in a backport.  I would suggest we use the same impl as InternalEngine here.


```java
    public Translog.Snapshot newChangesSnapshotFromTranslogFile(String source, long fromSeqNo, long toSeqNo, boolean requiredFullRange)
        throws IOException {
        return getTranslog().newSnapshot(fromSeqNo, toSeqNo, requiredFullRange);
    }
```